client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx

```diff
@@ -319,11 +324,18 @@ export const AppsSubmissionForm: React.FC = () => {
     setSteps(updatedSteps);
   }, [fields]);
 
-  const handleNextStep = useCallback(() => {
-    // setState({ ...state, ...data });
-    const nextPage = steps[current].nextPage;
-    nextPage && setCurrent(nextPage);
-  }, [current]);
+  // Only allow transition to next step, if the current step has
+  // no validation errors.
```

Comment:
future note: we should disable 'continue' and 'back' if we know the form is invalid

Reply:
sounds good, jira to track: [DES-2821](https://tacc-main.atlassian.net/browse/DES-2821)